builder/comp-builder.nix

```diff
@@ -196,187 +196,182 @@ let
       LOCALE_ARCHIVE = "${buildPackages.glibcLocales}/lib/locale/locale-archive";
     };
 
-in
-
-stdenv.lib.fix (drv:
-
-let
-  haddock = haddockBuilder {
-    inherit componentId component package flags commonConfigureFlags
-      commonAttrs revision setupGhcOptions doHaddock
-      doHoogle hyperlinkSource quickjump setupHaddockFlags
-      needsProfiling configFiles preHaddock postHaddock;
-
-    componentDrv = drv;
-  };
-in
-
-stdenv.mkDerivation (commonAttrs // {
-  pname = nameOnly;
-  inherit (package.identifier) version;
-
-  doCheck = false;
-  doCrossCheck = false;
-
-  inherit dontPatchELF dontStrip;
-
-  passthru = {
-    inherit (package) identifier;
-    config = component;
-    inherit configFiles executableToolDepends cleanSrc exeName;
-    env = shellWrappers;
-    profiled = self (drvArgs // { enableLibraryProfiling = true; });
-  } // lib.optionalAttrs (haskellLib.isLibrary componentId) ({
-      inherit haddock;
-      inherit (haddock) haddockDir; # This is null if `doHaddock = false`
-    } // lib.optionalAttrs (haddock ? doc) {
-      # `doHaddock = false` turns the doc of haddock output off
-      inherit (haddock) doc;
-    }
-  );
-
-  meta = {
-    homepage = package.homepage;
-    description = package.synopsis;
-    license =
-      let
-        license-map = import ../lib/cabal-licenses.nix lib;
-      in license-map.${package.license} or
-        (builtins.trace "WARNING: license \"${package.license}\" not found" license-map.LicenseRef-OtherLicense);
-    platforms = if platforms == null then stdenv.lib.platforms.all else platforms;
-  };
-
-  propagatedBuildInputs =
-       frameworks # Frameworks will be needed at link time
-    # Not sure why pkgconfig needs to be propagatedBuildInputs but
-    # for gi-gtk-hs it seems to help.
-    ++ builtins.concatLists pkgconfig;
+  drv =
+    let
```

Comment:
I don't think you need the nested let (and therefore the indentation) here.

Just continue the previous let with:
```
  haddock = ...
  drv = stdenv.mkDerivation ...
in drv
```